pkg/maintainer/wallet/redemptions.go

```diff
@@ -11,6 +12,43 @@ import (
 	"time"
 )
 
+func (wm *walletMaintainer) runRedemptionTask(ctx context.Context) error {
+	redemptionMaxSize, err := wm.chain.GetRedemptionMaxSize()
+	if err != nil {
+		return fmt.Errorf("failed to get deposit sweep max size: [%w]", err)
+	}
+
+	walletPublicKeyHash, redeemersOutputScripts, err := FindPendingRedemptions(
+		wm.chain,
+		[20]byte{},
+		redemptionMaxSize,
+	)
```

Comment:
`FindPendingRedemptions` in the current form stops searching when it finds a wallet with a pending redemption. I don't think this is enough. A malicious redeemer can easily make redemption requests to other wallets time out. I think we should - at a minimum - introduce a parameter that will limit the number of wallets that can process redemptions at the same time and not stop searching in `FindPendingRedemptions` until we reach the limit.

Maybe it's already on your radar for this PR, just making sure.

Reply:
Yep, this is captured in the task list of https://github.com/keep-network/keep-core/issues/3614